source/state_representation/test/tests/test_parameter_map.cpp

```diff
@@ -9,15 +10,20 @@ class TestParameterMap : public ParameterMap {
 public:
   explicit TestParameterMap(const ParameterInterfaceList& parameters) : ParameterMap(parameters) {}
 
+  using ParameterMap::assert_parameter_valid;
+
+  bool validate_called = false;
+
 private:
   void validate_and_set_parameter(const std::shared_ptr<ParameterInterface>& parameter) override {
-    assert_parameter_valid(parameter);
-    this->parameters_.insert_or_assign(parameter->get_name(), parameter);
+    validate_called = true;
+    ParameterMap::validate_and_set_parameter(parameter);
   }
 };
 
-TEST(ParameterMapTest, ParameterMap) {
-  auto map = ParameterMap();
+TEST(ParameterMapTest, ValidateSetParameter) {
+  // The TestParameterMap overrides `validate_and_set_parameter`, setting parameters should update the flag
+  auto map = TestParameterMap(ParameterInterfaceList());
   map.set_parameter(make_shared_parameter("int", 1));
```

Comment:
You didn't actually included the test for the validated property

```suggestion
  // The TestParameterMap overrides `validate_and_set_parameter`, setting parameters should update the flag
  auto map = TestParameterMap(ParameterInterfaceList());
  EXPECT_FALSE(map.validate_called);
  map.set_parameter(make_shared_parameter("int", 1));
  EXPECT_TRUE(map.validate_called);
  
```

Reply:
Wow yeah thats stupid, sorry that this goes on an on...